release notes/v0.40.0.md

```diff
@@ -0,0 +1,270 @@
+- [#2615](https://github.com/grafana/k6/pull/2615) Improve documentation of RegisterCallback due to feedback on how hard it was to understand how to use it properly.
+- [#2627](https://github.com/grafana/k6/pull/2627) Create distinct test state objects for the pre-init and run phases.
+- [#2635](https://github.com/grafana/k6/pull/2635) Drop License header in each file.
+- [#2636](https://github.com/grafana/k6/pull/2636) Add [SECURITY.md](https://github.com/grafana/k6/security/policy) with instructions how to report security issues responsibly.
+- [#2641](https://github.com/grafana/k6/pull/2641) Fix spelling of `lose`. Thanks @spazm!
+- Update to golangci-lint v1.47.2 and enable a bunch more linters. [#2609](https://github.com/grafana/k6/pull/2609), [#2611](https://github.com/grafana/k6/pull/2611). Also, drop obsolete configurations [#2619](https://github.com/grafana/k6/pull/2619).
+
+# Roadmap and future plans
+
+This section will talk about our plans for future versions. You will notice that two of the big ticket items are here again. This is mostly due to the sheer size of the work required on both and some unforeseen needed changes - most of them actually making this better in the long run. It also so happens that it is vacation season so the k6 team rightfully is taking some personal time away.
+
+## Native support for ECMAScript modules
+
+The native ESM support is coming. There is both under development [PR to k6](https://github.com/grafana/k6/pull/2563) and a branch that will become a PR to goja to add the support there. The k6 team is hopeful that this will land in the next version v0.41.0!
+
+It turned out that there were a lot more things to be added as functionality - dynamic import, and the [tc39](https://tc39.es/) group released the latest ECMAScript specification adding support of top-level await in modules. While neither k6 nor goja has support for the `async`/`await` syntax, yet, this changes significantly the internal of the change, which did require a not insignificant refactor.
+
+Additionally, as previously mentioned, there were a bunch of changes to goja, including adding class support which also needed to be integrated.
+
+A future breaking change () is that it will no longer be possible to use CommonJS style exports along import/export syntax in the *same* file.
+
+```javascript
+import http from "k6/http"
+
+exports.default = function() {} // this will start to error out
+```
+
+It will still be possible to import a CommonJS module and `require` an ES module or use `require` in an ES module. Having a dependency cycle mixing and matching CommonJS and ESM is also unlikely to work properly, but *might* do so in particular cases.
+
+This is really expected to have never been done by anyone as there isn't really much of a reason for this, but it is currently supported due to Babel transpiling everything to CommonJS behind the scenes.
+
```

Comment:
```suggestion
It turned out that much more functionality was to be added - dynamic import, and the [tc39](https://tc39.es/) group released the latest ECMAScript specification adding support of top-level await in modules. While neither k6 nor goja has support for the `async`/`await` syntax, yet, this significantly changes the internals of the change, which did require a not insignificant refactor.

Additionally, as previously mentioned, a bunch of changes happened to goja, including added class support, which also needed to be integrated.

A future breaking change () is that it will no longer be possible to use CommonJS-style exports along import/export syntax in the *same* file.
```

Reply:
No definitely not intentional! I think I overselected lines. Will revert, thanks.